src/mds/Server.cc

```diff
@@ -3779,7 +3779,6 @@ void Server::handle_client_setattr(MDRequestRef& mdr)
       mds->locker->drop_locks(mdr.get());
       mdr->drop_local_auth_pins();
       cur->add_waiter(CInode::WAIT_TRUNC, new C_MDS_RetryRequest(mdcache, mdr));
-      mds->mdlog->flush();
```

Comment:
On Tue, Jul 26, 2016 at 1:50 PM, Gregory Farnum <notifications@github.com>
wrote:

> In src/mds/Server.cc
> <https://github.com/ceph/ceph/pull/10393#discussion_r72191154>:
>
> > @@ -3779,7 +3779,6 @@ void Server::handle_client_setattr(MDRequestRef& mdr)
> >        mds->locker->drop_locks(mdr.get());
> >        mdr->drop_local_auth_pins();
> >        cur->add_waiter(CInode::WAIT_TRUNC, new C_MDS_RetryRequest(mdcache, mdr));
> > -      mds->mdlog->flush();
>
> I think we need to leave this flush in. Adding in an immediate flush once
> it's logged doesn't matter so much, but in this case isn't the client
> waiting for the update to be durable before it does anything?
>
Truncating inode has 3 three phrases.  Journal truncate start, truncate
objects and journal truncate finish. Taking inode's xlock trigger log flush
for the first phrase. MDCache::truncate_inode_finish() calls log flush for
the third phrase.

—

> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ceph/ceph/pull/10393/files/fdf98c10260546fdca13dbe2edc4445faccc13f5..0ecac2102ad64d7b2556ca4e9a48c53b12e2db43#r72191154>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACpZrxXsnuomid5StDxuJ2-kxhnF8ySGks5qZaAWgaJpZM4JShx2>
> .
>